### DISCUSSION TITLE
BlindTransfer with 482 Loop Detected help?

### DISCUSSION THREAD

**Discussion by PauloTroguilho** [sipsorcery-org/sipsorcery#441] [Q&A]
Hello,
I need help, I have an error trying to transfer to a sipproxy, I get the loop detected, my scenario is ... I get a call from siproxy, I answer and transfer to another extension of the sip proxy, sipuseragent answers and transfers.
Grateful!

**Comment by sipsorcery** [comment]
I'll need a SIP trace, either from Wireshark or from the sipsorcery logs.

To activate logs from a sipsorcery app call `sipTransport.EnableTraceLogs()`.

**Reply by PauloTroguilho** [reply]
Grateful for the return, follow SipTransport logs

2021-03-05 12:24:16.9605|DEBUG|PRACK sip:242424@192.168.20.56:5060 SIP/2.0
Via: SIP/2.0/UDP 192.168.20.190:5060;rport;branch=z9hG4bK-1c9211-6f9a9603-2445e4c7
To: "242424" <sip:242424@192.168.20.56:5060>;tag=GCZCHTOCAV
From: "Paulinho" <sip:11997407302@192.168.20.190:5060>;tag=15c286a0-be14a8c0-13c4-55022-1c9211-5f97c36-1c9211
Call-ID: 15ffa038-be14a8c0-13c4-55022-1c9211-63c57518-1c9211
CSeq: 2 PRACK
Max-Forwards: 70
User-Agent: Aspect SIP Routing Proxy 7.32.600.16021
Supported: 100rel,replaces
Content-Length: 0
RAck: 106567535 1 INVITE

 
2021-03-05 12:24:16.9605|INFO|INVITE 
2021-03-05 12:24:17.0186|DEBUG|Response sent: udp:192.168.20.56:5060--->udp:192.168.20.190:5060 
2021-03-05 12:24:17.0286|INFO|INVITE sip:242424@192.168.20.56:5060 SIP/2.0
Via: SIP/2.0/UDP 192.168.20.190:5060;rport=5060;branch=z9hG4bK-1c9211-6f9a94e6-77b99650;received=192.168.20.190
To: "242424" <sip:242424@192.168.20.56:5060>;tag=GCZCHTOCAV
From: "Paulinho" <sip:11997407302@192.168.20.190:5060>;tag=15c286a0-be14a8c0-13c4-55022-1c9211-5f97c36-1c9211
Call-ID: 15ffa038-be14a8c0-13c4-55022-1c9211-63c57518-1c9211
CSeq: 1 INVITE
Contact: <sip:UIPCORESEC73@192.168.20.190:5060>
Max-Forwards: 69
Record-Route: <sip:UIPCORESEC73@192.168.20.190:5060;transport=udp;lr>
User-Agent: Aspect SIP Routing Proxy 7.32.600.16021
Supported: 100rel,timer,replaces
Content-Length: 223
Content-Type: application/sdp
Session-Expires: 1800;refresher=uac

v=0
o=dcpsrv 1614957808 1614957808 IN IP4 192.168.20.101
s=call
c=IN IP4 192.168.20.101
t=0 0
m=audio 14632 RTP/AVP 8 96
a=rtpmap:8 PCMA/8000
a=ptime:20
a=rtpmap:96 telephone-event/8000
a=fmtp:96 0-15
a=sendrecv
 
2021-03-05 12:24:17.0379|DEBUG|SIP/2.0 200 Ok
Via: SIP/2.0/UDP 192.168.20.190:5060;rport=5060;branch=z9hG4bK-1c9211-6f9a9603-2445e4c7;received=192.168.20.190
To: "242424" <sip:242424@192.168.20.56:5060>;tag=GCZCHTOCAV
From: "Paulinho" <sip:11997407302@192.168.20.190:5060>;tag=15c286a0-be14a8c0-13c4-55022-1c9211-5f97c36-1c9211
Call-ID: 15ffa038-be14a8c0-13c4-55022-1c9211-63c57518-1c9211
CSeq: 2 PRACK
Allow: ACK, BYE, CANCEL, INFO, INVITE, NOTIFY, OPTIONS, PRACK, REFER, REGISTER, SUBSCRIBE
Content-Length: 0

 
2021-03-05 12:24:17.0616|DEBUG|Request received: udp:192.168.20.56:5060<---udp:192.168.20.190:5060 
2021-03-05 12:24:17.0767|DEBUG|ACK sip:192.168.20.56:5060 SIP/2.0
Via: SIP/2.0/UDP 192.168.20.190:5060;rport;branch=z9hG4bK-1c9212-6f9a96de-62f4a559
To: "242424" <sip:242424@192.168.20.56:5060>;tag=GCZCHTOCAV
From: "Paulinho" <sip:11997407302@192.168.20.190:5060>;tag=15c286a0-be14a8c0-13c4-55022-1c9211-5f97c36-1c9211
Call-ID: 15ffa038-be14a8c0-13c4-55022-1c9211-63c57518-1c9211
CSeq: 1 ACK
Contact: <sip:UIPCORESEC73@192.168.20.190:5060>
Max-Forwards: 70
User-Agent: Aspect SIP Routing Proxy 7.32.600.16021
Content-Length: 0

 
2021-03-05 12:24:19.8590|INFO|Paulinho.wav 
2021-03-05 12:24:23.4595|DEBUG|538fdf44829b4239bc9fd07ffcde155f - westus 
2021-03-05 12:24:23.4805|DEBUG|audioinput 
2021-03-05 12:24:23.7095|DEBUG|While 
2021-03-05 12:24:23.9066|INFO|
Session started event. 
2021-03-05 12:24:32.8262|DEBUG|RECOGNIZING: Text=sim 
2021-03-05 12:24:34.4390|DEBUG|RECOGNIZED INTENT: Text=Sim. 
2021-03-05 12:24:34.4576|DEBUG|    Intent Id: sim. 
2021-03-05 12:24:34.4931|DEBUG|    Language Understanding JSON: {
  "query": "sim",
  "topScoringIntent": {
    "intent": "Afirmativo",
    "score": 0.9840738
  },
  "entities": []
}. 
2021-03-05 12:24:34.5376|DEBUG|Fim While 
2021-03-05 12:24:39.0085|DEBUG|Request received: udp:192.168.20.56:5060<---udp:192.168.20.170:5060 
2021-03-05 12:24:39.0323|DEBUG|OPTIONS sip:rst_av@192.168.20.56:5060;transport=udp SIP/2.0
Via: SIP/2.0/UDP 192.168.20.170:5060;rport;branch=z9hG4bK-41ea40-17b0d92-17f91d94
To: <sip:rst_av@192.168.20.56:5060>;transport=udp
From: <sip:RSTUIPCORESEC@192.168.20.170>;tag=14e28dc0-aa14a8c0-13c4-55022-41ea40-859460a-41ea40
Call-ID: 14e28dc0-aa14a8c0-13c4-55022-41ea40-1d5698e5-41ea40
Max-Forwards: 70
Content-Length: 0

 
2021-03-05 12:24:39.0747|INFO|OPTIONS 
2021-03-05 12:24:39.1136|INFO|OPTIONS 
2021-03-05 12:24:39.1356|INFO|OPTIONS sip:rst_av@192.168.20.56:5060;transport=udp SIP/2.0
Via: SIP/2.0/UDP 192.168.20.170:5060;rport=5060;branch=z9hG4bK-41ea40-17b0d92-17f91d94;received=192.168.20.170
To: <sip:rst_av@192.168.20.56:5060>;transport=udp
From: <sip:RSTUIPCORESEC@192.168.20.170>;tag=14e28dc0-aa14a8c0-13c4-55022-41ea40-859460a-41ea40
Call-ID: 14e28dc0-aa14a8c0-13c4-55022-41ea40-1d5698e5-41ea40
Max-Forwards: 70
Content-Length: 0

 
2021-03-05 12:24:39.1898|DEBUG|Response sent: udp:192.168.20.56:5060--->udp:192.168.20.170:5060 
2021-03-05 12:24:39.2069|DEBUG|SIP/2.0 200 Ok
Via: SIP/2.0/UDP 192.168.20.170:5060;rport=5060;branch=z9hG4bK-41ea40-17b0d92-17f91d94;received=192.168.20.170
To: <sip:rst_av@192.168.20.56:5060>;transport=udp
From: <sip:RSTUIPCORESEC@192.168.20.170>;tag=14e28dc0-aa14a8c0-13c4-55022-41ea40-859460a-41ea40
Call-ID: 14e28dc0-aa14a8c0-13c4-55022-41ea40-1d5698e5-41ea40
CSeq: 100 OPTIONS
Allow: ACK, BYE, CANCEL, INFO, INVITE, NOTIFY, OPTIONS, PRACK, REFER, REGISTER, SUBSCRIBE
Content-Length: 0

 
2021-03-05 12:24:39.5015|DEBUG|Request received: udp:192.168.20.56:5060<---udp:192.168.20.170:5060 
2021-03-05 12:24:39.5108|DEBUG|OPTIONS sip:rst_av@192.168.20.56:5060;transport=udp SIP/2.0
Via: SIP/2.0/UDP 192.168.20.170:5060;rport;branch=z9hG4bK-41ea40-17b0d92-17f91d94
To: <sip:rst_av@192.168.20.56:5060>;transport=udp
From: <sip:RSTUIPCORESEC@192.168.20.170>;tag=14e28dc0-aa14a8c0-13c4-55022-41ea40-859460a-41ea40
Call-ID: 14e28dc0-aa14a8c0-13c4-55022-41ea40-1d5698e5-41ea40
Max-Forwards: 70
Content-Length: 0

 
2021-03-05 12:24:39.5255|INFO|OPTIONS 
2021-03-05 12:24:39.5255|INFO|OPTIONS 
2021-03-05 12:24:39.5452|INFO|OPTIONS sip:rst_av@192.168.20.56:5060;transport=udp SIP/2.0
Via: SIP/2.0/UDP 192.168.20.170:5060;rport=5060;branch=z9hG4bK-41ea40-17b0d92-17f91d94;received=192.168.20.170
To: <sip:rst_av@192.168.20.56:5060>;transport=udp
From: <sip:RSTUIPCORESEC@192.168.20.170>;tag=14e28dc0-aa14a8c0-13c4-55022-41ea40-859460a-41ea40
Call-ID: 14e28dc0-aa14a8c0-13c4-55022-41ea40-1d5698e5-41ea40
Max-Forwards: 70
Content-Length: 0

 
2021-03-05 12:24:39.5680|DEBUG|Response sent: udp:192.168.20.56:5060--->udp:192.168.20.170:5060 
2021-03-05 12:24:39.5680|DEBUG|SIP/2.0 200 Ok
Via: SIP/2.0/UDP 192.168.20.170:5060;rport=5060;branch=z9hG4bK-41ea40-17b0d92-17f91d94;received=192.168.20.170
To: <sip:rst_av@192.168.20.56:5060>;transport=udp
From: <sip:RSTUIPCORESEC@192.168.20.170>;tag=14e28dc0-aa14a8c0-13c4-55022-41ea40-859460a-41ea40
Call-ID: 14e28dc0-aa14a8c0-13c4-55022-41ea40-1d5698e5-41ea40
CSeq: 100 OPTIONS
Allow: ACK, BYE, CANCEL, INFO, INVITE, NOTIFY, OPTIONS, PRACK, REFER, REGISTER, SUBSCRIBE
Content-Length: 0

 
2021-03-05 12:24:40.5173|DEBUG|Request received: udp:192.168.20.56:5060<---udp:192.168.20.170:5060 
2021-03-05 12:24:40.5394|DEBUG|OPTIONS sip:rst_av@192.168.20.56:5060;transport=udp SIP/2.0
Via: SIP/2.0/UDP 192.168.20.170:5060;rport;branch=z9hG4bK-41ea40-17b0d92-17f91d94
To: <sip:rst_av@192.168.20.56:5060>;transport=udp
From: <sip:RSTUIPCORESEC@192.168.20.170>;tag=14e28dc0-aa14a8c0-13c4-55022-41ea40-859460a-41ea40
Call-ID: 14e28dc0-aa14a8c0-13c4-55022-41ea40-1d5698e5-41ea40
Max-Forwards: 70
Content-Length: 0

 
2021-03-05 12:24:40.6032|INFO|OPTIONS 
2021-03-05 12:24:40.6405|INFO|OPTIONS 
2021-03-05 12:24:40.6595|INFO|OPTIONS sip:rst_av@192.168.20.56:5060;transport=udp SIP/2.0
Via: SIP/2.0/UDP 192.168.20.170:5060;rport=5060;branch=z9hG4bK-41ea40-17b0d92-17f91d94;received=192.168.20.170
To: <sip:rst_av@192.168.20.56:5060>;transport=udp
From: <sip:RSTUIPCORESEC@192.168.20.170>;tag=14e28dc0-aa14a8c0-13c4-55022-41ea40-859460a-41ea40
Call-ID: 14e28dc0-aa14a8c0-13c4-55022-41ea40-1d5698e5-41ea40
Max-Forwards: 70
Content-Length: 0

 
2021-03-05 12:24:40.6850|DEBUG|Response sent: udp:192.168.20.56:5060--->udp:192.168.20.170:5060 
2021-03-05 12:24:40.6986|DEBUG|SIP/2.0 200 Ok
Via: SIP/2.0/UDP 192.168.20.170:5060;rport=5060;branch=z9hG4bK-41ea40-17b0d92-17f91d94;received=192.168.20.170
To: <sip:rst_av@192.168.20.56:5060>;transport=udp
From: <sip:RSTUIPCORESEC@192.168.20.170>;tag=14e28dc0-aa14a8c0-13c4-55022-41ea40-859460a-41ea40
Call-ID: 14e28dc0-aa14a8c0-13c4-55022-41ea40-1d5698e5-41ea40
CSeq: 100 OPTIONS
Allow: ACK, BYE, CANCEL, INFO, INVITE, NOTIFY, OPTIONS, PRACK, REFER, REGISTER, SUBSCRIBE
Content-Length: 0

 
2021-03-05 12:24:41.2784|INFO|Transferindo para sip:242425@192.168.20.188:5060=========================> 
2021-03-05 12:24:41.3146|INFO|Transferindo para com User-to-usersip:242425@192.168.20.188:5060Paulinho 
2021-03-05 12:24:41.3545|DEBUG|Request sent: udp:192.168.20.56:5060--->udp:192.168.20.190:5060 
2021-03-05 12:24:41.3890|DEBUG|REFER sip:UIPCORESEC73@192.168.20.190:5060 SIP/2.0
Via: SIP/2.0/UDP 192.168.20.56:5060;branch=z9hG4bK7eb416a6511144b6814568d9f4733d6f;rport
To: "Paulinho" <sip:11997407302@192.168.20.190:5060>;tag=15c286a0-be14a8c0-13c4-55022-1c9211-5f97c36-1c9211
From: "242424" <sip:242424@192.168.20.56:5060>;tag=GCZCHTOCAV
Call-ID: 15ffa038-be14a8c0-13c4-55022-1c9211-63c57518-1c9211
CSeq: 2 REFER
Contact: <sip:192.168.20.56:5060>
Max-Forwards: 70
Route: <sip:UIPCORESEC73@192.168.20.190:5060;transport=udp;lr>
Supported: norefersub
Content-Length: 0
Refer-To: sip:242425@192.168.20.188:5060
User-to-User: TESTE RESOLUTEPaulinho11997407302; enconding=HEX

 
2021-03-05 12:24:41.4178|DEBUG|Response received: udp:192.168.20.56:5060<---udp:192.168.20.190:5060 
2021-03-05 12:24:41.4178|DEBUG|SIP/2.0 202 Accepted
Via: SIP/2.0/UDP 192.168.20.56:5060;rport=5060;branch=z9hG4bK7eb416a6511144b6814568d9f4733d6f
To: "Paulinho" <sip:11997407302@192.168.20.190:5060>;tag=15c286a0-be14a8c0-13c4-55022-1c9211-5f97c36-1c9211
From: "242424" <sip:242424@192.168.20.56:5060>;tag=GCZCHTOCAV
Call-ID: 15ffa038-be14a8c0-13c4-55022-1c9211-63c57518-1c9211
CSeq: 2 REFER
Contact: <sip:UIPCORESEC73@192.168.20.190:5060;transport=UDP>
Supported: timer,replaces
Content-Length: 0

 
2021-03-05 12:24:41.4522|DEBUG|Request received: udp:192.168.20.56:5060<---udp:192.168.20.190:5060 
2021-03-05 12:24:41.4522|INFO|Desligando Tranferiu 
2021-03-05 12:24:41.4617|DEBUG|NOTIFY sip:192.168.20.56:5060 SIP/2.0
Via: SIP/2.0/UDP 192.168.20.190:5060;rport;branch=z9hG4bK-1c922a-6f9af66d-29cae470
To: "242424" <sip:242424@192.168.20.56:5060>;tag=GCZCHTOCAV
From: "Paulinho" <sip:11997407302@192.168.20.190:5060>;tag=15c286a0-be14a8c0-13c4-55022-1c9211-5f97c36-1c9211
Call-ID: 15ffa038-be14a8c0-13c4-55022-1c9211-63c57518-1c9211
CSeq: 3 NOTIFY
Contact: <sip:UIPCORESEC73@192.168.20.190:5060>
Max-Forwards: 70
Supported: 100rel,timer,replaces
Content-Length: 22
Content-Type: message/sipfrag
Event: refer;id=1
Subscription-State: active;expires=50

SIP/2.0 100 Trying

 
2021-03-05 12:24:41.4916|INFO|Saiu 
2021-03-05 12:24:41.4916|DEBUG|Request received: udp:192.168.20.56:5060<---udp:192.168.20.190:5060 
2021-03-05 12:24:41.4916|INFO|NOTIFY 
2021-03-05 12:24:41.5114|DEBUG|NOTIFY sip:192.168.20.56:5060 SIP/2.0
Via: SIP/2.0/UDP 192.168.20.190:5060;rport;branch=z9hG4bK-1c922a-6f9af66f-60a6822f
To: "242424" <sip:242424@192.168.20.56:5060>;tag=GCZCHTOCAV
From: "Paulinho" <sip:11997407302@192.168.20.190:5060>;tag=15c286a0-be14a8c0-13c4-55022-1c9211-5f97c36-1c9211
Call-ID: 15ffa038-be14a8c0-13c4-55022-1c9211-63c57518-1c9211
CSeq: 4 NOTIFY
Contact: <sip:UIPCORESEC73@192.168.20.190:5060>
Max-Forwards: 70
Supported: 100rel,timer,replaces
Content-Length: 22
Content-Type: message/sipfrag
Event: refer;id=1
Subscription-State: active;expires=50

SIP/2.0 100 Trying

 
2021-03-05 12:24:41.5114|INFO|NOTIFY 
2021-03-05 12:24:41.5313|DEBUG|Request sent: udp:192.168.20.56:5060--->udp:192.168.20.190:5060 
2021-03-05 12:24:41.5548|DEBUG|Request received: udp:192.168.20.56:5060<---udp:192.168.20.190:5060 
2021-03-05 12:24:41.5548|INFO|NOTIFY sip:192.168.20.56:5060 SIP/2.0
Via: SIP/2.0/UDP 192.168.20.190:5060;rport=5060;branch=z9hG4bK-1c922a-6f9af66d-29cae470;received=192.168.20.190
To: "242424" <sip:242424@192.168.20.56:5060>;tag=GCZCHTOCAV
From: "Paulinho" <sip:11997407302@192.168.20.190:5060>;tag=15c286a0-be14a8c0-13c4-55022-1c9211-5f97c36-1c9211
Call-ID: 15ffa038-be14a8c0-13c4-55022-1c9211-63c57518-1c9211
CSeq: 3 NOTIFY
Contact: <sip:UIPCORESEC73@192.168.20.190:5060>
Max-Forwards: 70
Supported: 100rel,timer,replaces
Content-Length: 22
Content-Type: message/sipfrag
Event: refer;id=1
Subscription-State: active;expires=50

SIP/2.0 100 Trying

 
2021-03-05 12:24:41.5673|INFO|NOTIFY 
2021-03-05 12:24:41.5673|DEBUG|BYE sip:UIPCORESEC73@192.168.20.190:5060 SIP/2.0
Via: SIP/2.0/UDP 192.168.20.56:5060;branch=z9hG4bK8768017e27094d7bbe0018b5e7590d87;rport
To: "Paulinho" <sip:11997407302@192.168.20.190:5060>;tag=15c286a0-be14a8c0-13c4-55022-1c9211-5f97c36-1c9211
From: "242424" <sip:242424@192.168.20.56:5060>;tag=GCZCHTOCAV
Call-ID: 15ffa038-be14a8c0-13c4-55022-1c9211-63c57518-1c9211
CSeq: 3 BYE
Max-Forwards: 70
Route: <sip:UIPCORESEC73@192.168.20.190:5060;transport=udp;lr>
Content-Length: 0

 
2021-03-05 12:24:41.5872|DEBUG|NOTIFY sip:192.168.20.56:5060 SIP/2.0
Via: SIP/2.0/UDP 192.168.20.190:5060;rport;branch=z9hG4bK-1c922a-6f9af677-1add7fbb
To: "242424" <sip:242424@192.168.20.56:5060>;tag=GCZCHTOCAV
From: "Paulinho" <sip:11997407302@192.168.20.190:5060>;tag=15c286a0-be14a8c0-13c4-55022-1c9211-5f97c36-1c9211
Call-ID: 15ffa038-be14a8c0-13c4-55022-1c9211-63c57518-1c9211
CSeq: 5 NOTIFY
Contact: <sip:UIPCORESEC73@192.168.20.190:5060>
Max-Forwards: 70
Supported: 100rel,timer,replaces
Content-Length: 29
Content-Type: message/sipfrag
Event: refer;id=1
Subscription-State: terminated;reason=noresource

SIP/2.0 482 Loop Detected

 
2021-03-05 12:24:41.6074|INFO|NOTIFY 
2021-03-05 12:24:41.6450|DEBUG|Response received: udp:192.168.20.56:5060<---udp:192.168.20.190:5060 
2021-03-05 12:24:41.6450|INFO|NOTIFY 
2021-03-05 12:24:41.6522|DEBUG|SIP/2.0 200 OK
Via: SIP/2.0/UDP 192.168.20.56:5060;rport=5060;branch=z9hG4bK8768017e27094d7bbe0018b5e7590d87
To: "Paulinho" <sip:11997407302@192.168.20.190:5060>;tag=15c286a0-be14a8c0-13c4-55022-1c9211-5f97c36-1c9211
From: "242424" <sip:242424@192.168.20.56:5060>;tag=GCZCHTOCAV
Call-ID: 15ffa038-be14a8c0-13c4-55022-1c9211-63c57518-1c9211
CSeq: 3 BYE
Supported: timer,replaces
Content-Length: 0

 
2021-03-05 12:24:41.6657|INFO|NOTIFY 
2021-03-05 12:24:41.6522|INFO|NOTIFY sip:192.168.20.56:5060 SIP/2.0
Via: SIP/2.0/UDP 192.168.20.190:5060;rport=5060;branch=z9hG4bK-1c922a-6f9af66f-60a6822f;received=192.168.20.190
To: "242424" <sip:242424@192.168.20.56:5060>;tag=GCZCHTOCAV
From: "Paulinho" <sip:11997407302@192.168.20.190:5060>;tag=15c286a0-be14a8c0-13c4-55022-1c9211-5f97c36-1c9211
Call-ID: 15ffa038-be14a8c0-13c4-55022-1c9211-63c57518-1c9211
CSeq: 4 NOTIFY
Contact: <sip:UIPCORESEC73@192.168.20.190:5060>
Max-Forwards: 70
Supported: 100rel,timer,replaces
Content-Length: 22
Content-Type: message/sipfrag
Event: refer;id=1
Subscription-State: active;expires=50

SIP/2.0 100 Trying

 
2021-03-05 12:24:41.7276|INFO|NOTIFY sip:192.168.20.56:5060 SIP/2.0
Via: SIP/2.0/UDP 192.168.20.190:5060;rport=5060;branch=z9hG4bK-1c922a-6f9af677-1add7fbb;received=192.168.20.190
To: "242424" <sip:242424@192.168.20.56:5060>;tag=GCZCHTOCAV
From: "Paulinho" <sip:11997407302@192.168.20.190:5060>;tag=15c286a0-be14a8c0-13c4-55022-1c9211-5f97c36-1c9211
Call-ID: 15ffa038-be14a8c0-13c4-55022-1c9211-63c57518-1c9211
CSeq: 5 NOTIFY
Contact: <sip:UIPCORESEC73@192.168.20.190:5060>
Max-Forwards: 70
Supported: 100rel,timer,replaces
Content-Length: 29
Content-Type: message/sipfrag
Event: refer;id=1
Subscription-State: terminated;reason=noresource

SIP/2.0 482 Loop Detected

**Reply by PauloTroguilho** [reply]
I received a call from "Paulinho" sip: 11997407302@192.168.20.190: 5060 to sip: 242424@192.168.20.56: 5060, I answered
and tried to transfer to sip: 242425@192.168.20.188: 5060

activeAgent.BlindTransfer method used, when i receive from another location if not from sipproxy i can transfer normal, only when i receive a call from sipproxy and i will transfer to another extension of the same siproxy i get the loop

**Reply by PauloTroguilho** [reply]
I removed Rota: sip: UIPCORESEC73@192.168.20.190: 5060; transport = udp; lr, I saw that if he has a Route he sends it to the route instead of sending it to the Refer-To, but an error persists.

**Comment by sipsorcery** [comment]
The INVITE OK response is missing. It can contain some important headers that dictate the REFER request construction. Assuming the OK response is stock standard then the REFER request from the application using the sipsorcery library looks normal except that the `Route` header are to the same host:

URI: `sip:UIPCORESEC73@192.168.20.190:5060`
Route: `sip:UIPCORESEC73@192.168.20.190:5060;transport=udp;lr`

This means the SIP server listening on `192.168.20.190:5060` is being asked to:

 - Accept a REFER request to `sip:UIPCORESEC73@192.168.20.190:5060;transport=udp;lr`
 - Then forward it to itself on `sip:UIPCORESEC73@192.168.20.190:5060`.

Technically that's permitted but given the `Loop Detected` response the suspicion would be that the SIP server on `192.168.20.190:5060` is not happy about being asked to act as both a route and a destination.

While the INVITE OK response would shed a bit more light my first suggestion would be to remove any `Record-Route` header that the  `192.168.20.190:5060` server is setting on the INVITE response.

If you've already tried that and that my next guess is due to the indication it looks like you might be running the at least one of the SIP servers on AWS. If that's the case then you may be using the `Record-Route` header to deal with NAT. For example an attempt could be being made to configure the `192.168.20.190:5060` SIP server to handle both Internet and private traffic. That type of configuration can get complicated but it will depend on the type of SIP server. It doesn't seem like that server is using the sipsorcery library.

**Comment by PauloTroguilho** [comment]
Thank you very much for the information, I help a lot, really the server is rejecting it. Congratulations on the library, very good!
